src/components/Settings.tsx

```diff
@@ -0,0 +1,101 @@
+import React from "react";
+import tw, { styled } from "twin.macro";
+import Button from "./Button";
+import UnstyledHeading from "./Heading";
+
+import { useConnection } from "../hooks";
+
+type PingProps = { isConnected: boolean };
+const PingOuter = tw.div`
+  animate-ping absolute inline-flex h-full w-full rounded-full bg-primary opacity-75
+`;
+const PingInner = styled.div<PingProps>`
+  ${tw`relative inline-flex rounded-full w-1 h-1 bg-primary`};
+  ${({ isConnected }) => (isConnected ? tw`bg-primary` : tw`bg-gray-darker`)};
+`;
+const PingWrapper = tw.div`relative mr-2 flex w-1 w-1`;
+const Ping: React.FC<PingProps> = ({ isConnected }) => (
+  <PingWrapper>
+    {isConnected && <PingOuter />}
+    <PingInner isConnected={isConnected} />
+  </PingWrapper>
+);
+
+type SettingsProps = {
+  onCancel: () => void;
+  onSave: () => void;
+};
+const Settings: React.FC<SettingsProps> = ({ onCancel, onSave }) => {
+  const {
+    address,
+    onboard,
+    isConnected: isConnectedToWeb3,
+    disconnect,
+  } = useConnection();
+  const [hasRemovedWallet, setHasRemovedWallet] = React.useState(false);
+  const handleRemove = () => {
+    setHasRemovedWallet(true);
```

Comment:
what do you think about just removing wallet when the user clicks remove, rather than how it is now where you click remove -> save